controllers/workspace/devworkspace_controller.go

```diff
@@ -240,6 +241,22 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		HttpClient:         httpClient,
 	}
 
+	// TODO: Use merge code rather than deepCopy which overwrites?
+	if len(workspace.Spec.Template.Components) == 0 && config.Workspace.DefaultTemplate != nil {
```

Comment:
This should be broken into a function, probably in a package within `pkg/library`